### DISCUSSION TITLE
Group linting jobs under one common CI workflow?

### DISCUSSION THREAD

**Discussion by Wirone** [PHP-CS-Fixer/PHP-CS-Fixer#7485] [RFC]
I find `YAML Validation / Validate YAML` redundant. Additionally, new workflow was added for Sphinx validation. What I see here as a common pattern is linting, so maybe it's possible to group them under one `Lint` workflow, then jobs can make it more precise:

- `Lint / YAML`
- `Lint / reStructuredText`
- `Lint / Markdown` (not existing now, but can be added)
- `Lint / Dockerfile` (not existing now, but can be added)
- `Lint / Pull Request name`
etc.

I am more of a Gitlab user and this is something I would do there, several `lint:*` jobs.

**Comment by szepeviktor** [comment]
Stuffing all these checks under one workflow shows low importance.
If you care about every byte in the repository look at these workflows: https://github.com/szepeviktor/byte-level-care/tree/master/.github/workflows
Now everyone may agree: it is an overkill to care about every byte.

**Reply by szepeviktor** [reply]
> Stuffing all these checks under one workflow shows

that you don't have a dedicated person for QA

**Reply by Wirone** [reply]
For me this is only the way of organising stuff and may be (probably _is_) subjective. It has nothing to do with "caring (or not) about every byte" - you either have the check or you don't have it, how it's named and where it's placed is not that important in terms of QA. It's important in terms of readability and familiarity, though.

**Reply by szepeviktor** [reply]
I mean I do like to live in a workflow jungle 🐯 🌴

**Reply by keradus** [reply]
@Wirone , introduce groups when we have elements to group. for 2 jobs, don't bother with grouping them

**Reply by Wirone** [reply]
@keradus that's why I pointed out at least 2 linters that we could add to the workflow 😉.

**Comment by keradus** [comment]
> I find `YAML Validation / Validate YAML` redundant

redundant with? `yamllint` is called only in that single job

**Reply by Wirone** [reply]
Redundant, because the workflow and the job have similar name: `YAML Validation` vs `Validate YAML`.

**Comment by keradus** [comment]
> Group linting jobs under one common CI workflow?
> Lint / Pull Request name

**Approach 1**: ok, let's covert everything to `Lint / sth`
why? https://en.wikipedia.org/wiki/Lint_(software)
- SCA are linting by definition, eg phpstan or whitespaces checks
- actually, running `php-cs-fixer check` is also linting
- if we follow definition on Wiki (not best, but Oxford dict not helping here!) - even detecting bugs is form of linting (hm... PHPUnit?)

So... everything should be `Lint / *`. So, we can skip the common prefix of `Lint / ` and **keep the status quo**

-----

**Approach 2**: Effort Matrix exercise.
What is the value?
I see effort _for discussion_ (which IMHO already took more than value of proposal, if implemented), effort for _implementing_ (I understand you are OK to take it on you, sure - but let's continue reading) and effort for _reviewing/testing_ (which cannot be you).

I would suggest to compare the outcome of value/effort result to other awaiting tasks and prioritised accordingly.
With results of prioritisation, it (in my head) goes to very bottom of topics, aka we **keep the status quo**

**Comment by szepeviktor** [comment]
@Wirone I've told you "linting" has a personal meaning, is not objective.
Coding standards, syntax check are concrete things.
